### ISSUE TITLE
[serve] Re-submitting the same config multiple times always causes the application to be redeployed

### ISSUE THREAD

**Issue by edoakes** [ray-project/ray#34096]
`app.py`:
```python
from ray import serve

@serve.deployment
class A:
    pass

a = A.bind()
```

`config.yaml`:
```yaml
import_path: app:a
```

Repro (note how the timestamp changed):
```bash
$ ray start --head
$ serve deploy config.py
$ serve status
name: ''
app_status:
  status: RUNNING
  message: ''
  deployment_timestamp: 1680704706.176568
deployment_statuses:
- name: A
  status: HEALTHY
  message: ''

$ serve deploy config.yaml
$ serve status
name: ''
app_status:
  status: RUNNING
  message: ''
  deployment_timestamp: 1680704862.105747
deployment_statuses:
- name: A
  status: HEALTHY
  message: ''
```

Controller logs confirm it:
```
INFO 2023-04-05 09:30:15,109 controller 66710 application_state.py:325 - Received new config deployment for  request. Cancelling previous request.
INFO 2023-04-05 09:30:15,765 controller 66710 deployment_state.py:1265 - Stopping 1 replicas of deployment 'A' with outdated versions.
INFO 2023-04-05 09:30:17,947 controller 66710 deployment_state.py:1376 - Adding 1 replica to deployment 'A'.
```

**Comment by edoakes** [comment]
FYI I also tried this including the `deployments` list and got the same behavior (to dedup with the other known issue).

**Comment by zcin** [comment]
Oh also, currently the timestamp always changes as long as `serve deploy` is called, regardless of if any deployments redeployed.

**Comment by edoakes** [comment]
@zcin I tested with the deployment listed both times and saw the same behavior

**Comment by zcin** [comment]
Hm that's weird, I can't reproduce on my local env. The deployment timestamp gets updated, but I don't see controller logs that indicate the replica is being stopped and re-added.

**Comment by edoakes** [comment]
Let me try again

**Comment by edoakes** [comment]
I'm seeing the same with my config:
```
import_path: app:a
deployments:
  - name: A
    num_replicas: 2
```

Closing issue since it appears to be a dup of the other you're working on
